CorsixTH/Lua/app.lua

```diff
@@ -1709,9 +1716,10 @@ function App:checkForUpdates()
 
   local update_table = loadstring_envcall(update_body, "@updatechecker"){}
   local changelog = update_table["changelog_" .. default_language]
-  local new_version = update_table["major"] .. '.' .. update_table["minor"] .. update_table["revision"]
+  local new_version = update_table["major"] .. '.' .. update_table["minor"] .. "." .. update_table["revision"]
+  local new_version_major, new_version_minor, new_version_revision = extractVersionNumericals(new_version)
 
-  if (new_version <= current_version) then
+  if (new_version_major <= current_version_major) and (new_version_minor <= current_version_minor) and (new_version_revision <= current_version_revision) then
```

Comment:
So I have version `3.7.0` while we just released `4.1.0`. Phew, I am up-to-date.

Reply:
Hmm, maybe I am not, I am getting very confused between `new` and `current` here.

Never mind, it works as long as I am not running a future version I think.

Reply:
Did I make an error somewhere? I really don't like how I wrote this if statement just in order to save space.
Realising we only run this function once per run -- perhaps more clarity over saving space is better

Reply:
I think it is correct. It's just a compact way to compute the result.

What I did was attempting to break it with the `7` being larger than the `1`, but that example is irrelevant since with normal version numbers you then also must have the `3` and the `4`, since I cannot have `4.7.0` if `4.1.0` is the newest release.
The `3` versus `4` comparison already decides that the version is obsolete and the `7` versus `1` isn't relevant anymore.